### PR TITLE
Supported `tf.float16` in `tf.decode_raw`.

### LINKED ISSUE #4782: tf.decode_raw should support tf.float16
Hi!

It should be trivial to add, I guess.

### PR DESCRIPTION
**Pull request by haosdent**
This fixes #4782.

### CODE REVIEW

**Comment by haosdent** [comment]
@benoitsteiner Would you help review this when your available? Thank you very much.

File: tensorflow/core/ops/parsing_ops.cc [hunk 1]
```diff
@@ -27,7 +27,7 @@ using shape_inference::ShapeHandle;
 REGISTER_OP("DecodeRaw")
     .Input("bytes: string")
     .Output("output: out_type")
-    .Attr("out_type: {float,double,int32,uint8,int16,int8,int64}")
+    .Attr("out_type: {float16,float,double,int32,uint8,int16,int8,int64}")
```
Comment (benoitsteiner) [inline tensorflow/core/ops/parsing_ops.cc]: We call 'float16' 'half' instead in tensorflow. See for example the definition of the BatchMatMul op in tensorflow/core/ops/math_ops.cc

File: tensorflow/python/kernel_tests/decode_raw_op_test.py [hunk 2]
```diff
@@ -57,5 +58,19 @@ def testToInt16(self):
           "size of int16"):
         decode.eval(feed_dict={in_bytes: ["123", "456"]})
 
+  def testToFloat16(self):
+    with self.test_session():
+      in_bytes = tf.placeholder(tf.string, shape=[None])
+      decode = tf.decode_raw(in_bytes, out_type=tf.float16)
+      self.assertEqual([None, None], decode.get_shape().as_list())
+
+      expected_result = np.matrix([[1, 2, 3, 4]], dtype=np.float16)
```
Comment (benoitsteiner) [inline tensorflow/python/kernel_tests/decode_raw_op_test.py]: I'm nitpicking here but you might want to consider feeding in a few negative values as well.

**Comment by haosdent** [comment]
@benoitsteiner Thanks a lot for your kindly helps, would you help review this again? Thank you in advance!

**Review by benoitsteiner — APPROVED** [review]

**Comment by benoitsteiner** [comment]
@tensorflow-jenkins test this please

**Pull request merged**